File: SubscriptionApi/SubscriptionApi/Subscriptions/Subscription.cs
using SubscriptionApi.Dto;

namespace SubscriptionApi.Subscriptions
{
  public class Subscription
  {
    private readonly NewSubscriptionParametersDto _parameters;

    public Subscription(NewSubscriptionParametersDto parameters)
    {
      _parameters = parameters;
    }

    public bool Has(string subscriptionId)
    {
      return _parameters.SubscriptionId == subscriptionId;
    }

    public void ForceExpiry()
    {
      throw new System.NotImplementedException();
    }

    public void ScheduleExpiry()
    {
      throw new System.NotImplementedException();
    }
  }
}

File: Java/factories/_01_encapsulation_of_global_context/src/main/java/subscriptions/Subscription.java
package subscriptions;

import dto.NewSubscriptionParametersDto;

public class Subscription {
    private final NewSubscriptionParametersDto parameters;

    public Subscription(NewSubscriptionParametersDto parameters) {
        this.parameters = parameters;
    }

    public boolean has(String subscriptionId) {
        return parameters.subscriptionId == subscriptionId;
    }

    public void forceExpiry() {
        throw new RuntimeException("not implemented");
    }

    public void scheduleExpiry() {
        throw new RuntimeException("not implemented");
    }
}
